### PR TITLE
feat(installer): native menus, credits, and i18n

### PR DESCRIPTION
**Pull request by FaberVi**
## Summary
- Add native app menus, tray, and About credits for the Tauri installer.
- Introduce Rust i18n plumbing and locale-aware menu labels.
- Fix dashboard open flow and related command/window handling.

## Test plan
- [ ] `npm run tauri dev` in `installer/` (or desktop build)
- [ ] Verify menu/tray actions and About dialog credits
- [ ] Open dashboard from menu after setup

### CODE REVIEW

**Comment by rahilp** [comment]
Heads-up @FaberVi — I switched the base of this PR from `main` to the `v2.2` release branch, where work is landing ahead of the next release. It reaches `main` when `v2.2` ships.

Nothing needed from you: `v2.2` is just `main` plus a few commits, so the diff is unchanged and it still merges cleanly. Thanks!

File: installer/src-tauri/src/lib.rs [hunk 1]
```diff
@@ -5,35 +5,47 @@
...
 mod secure_store;
 mod version;
 mod windows;
 mod worker_bundle;
 
+use app_menus::{build_menu_items, build_tray_items, install_app_menu, install_tray, AppMenus};
 use commands::SetupSession;
-use tauri::menu::{Menu, MenuBuilder, MenuItem, SubmenuBuilder};
-use tauri::tray::TrayIconBuilder;
+use i18n::{self, AppLocale, Key};
```
Comment (rahilp) [inline installer/src-tauri/src/lib.rs]:
```
error[E0255]: the name `i18n` is defined multiple times
  --> src/lib.rs:24:12
warning: unused import: `self`
```

`mod i18n;` is declared above, which already binds `i18n` in this scope. `use i18n::{self, ...}` then tries to bind the same name a second time.

The `self` is also unused — everything referenced goes through the explicit `AppLocale` / `Key` imports. Dropping it should resolve both the error and the warning:

```rust
use i18n::{AppLocale, Key};
```

File: installer/src-tauri/src/app_menus.rs [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+{
+    TrayIconBuilder::with_id("second-brain-tray")
+        .icon(
+            app.default_window_icon()
+                .expect("bundled icon")
+                .clone(),
+        )
+        .menu(tray_menu)
+        .show_menu_on_left_click(true)
+        .on_menu_event(on_menu_event)
```
Comment (rahilp) [inline installer/src-tauri/src/app_menus.rs:263]:
```
error[E0277]: expected a `Fn(&AppHandle<_>, tauri::menu::MenuEvent)` closure, found `F`
  --> src/app_menus.rs:263:24
```

`on_menu_event` wants something callable as `Fn(&AppHandle<R>, MenuEvent)`, and the generic `F` reaching this call site isn't constrained to that.

Whichever function or generic parameter is being threaded through here needs the bound spelled out on the enclosing signature, e.g.

```rust
where F: Fn(&AppHandle<R>, MenuEvent) + Send + Sync + 'static
```

Worth double-checking the argument order and the reference on the first parameter — `&AppHandle`, not `AppHandle` — since that's a common source of this exact mismatch.

Related: `Manager` is imported at line 9 but unused, which suggests this area was mid-refactor.

File: installer/src-tauri/src/commands.rs [hunk 3]
```diff
@@ -592,38 +636,71 @@ pub fn open_dashboard_integrations(
...
+    let locale = Locale::parse(&locale).ok_or_else(|| "Invalid locale".to_string())?;
+    if let Ok(config) = app.path().app_config_dir() {
+        let _ = i18n::write_stored_locale(&config, locale);
+    }
+    if let Some(state) = app.try_state::<AppLocale>() {
+        state.set(locale);
+    }
+    if let Some(menus) = app.try_state::<AppMenus>() {
+        menus.apply_locale(locale);
+        menus.rebuild_tray_menu(&app)?;
```
Comment (rahilp) [inline installer/src-tauri/src/commands.rs]:
```
error[E0277]: `?` couldn't convert the error to `std::string::String`
  --> src/commands.rs:703:38
  = note: the trait `From<tauri::Error>` is not implemented for `String`
```

`rebuild_tray_menu` returns a `tauri::Error`, but this function returns `Result<_, String>`, and there's no `From<tauri::Error> for String` — so bare `?` can't bridge them.

The codebase convention elsewhere is to map explicitly:

```rust
menus.rebuild_tray_menu(&app).map_err(|e| e.to_string())?;
```

Worth a scan for the same pattern nearby — `apply_locale` on the line above returns `()` so it's fine, but any other `?` on a `tauri::Error` in a `String`-error function will hit this too.

**Review by rahilp — CHANGES_REQUESTED** [review]
Thank you for this, @FaberVi — native menus, a credits surface and a full i18n layer is a lot of ground, and the TypeScript side builds and runs cleanly. I want to flag up front that the problem below is one **our CI should have caught for you and didn't**, so please read this as a gap on our side that you walked into rather than something you should have spotted unaided.

## The blocker: the Rust crate doesn't compile

`cargo check --manifest-path installer/src-tauri/Cargo.toml` fails with three errors:

```
lib.rs:24:12        E0255  the name `i18n` is defined multiple times
app_menus.rs:263:24 E0277  expected a `Fn(&AppHandle<_>, tauri::menu::MenuEvent)` closure, found `F`
commands.rs:703:38  E0277  `?` couldn't convert the error to `String`
                           (`From<tauri::Error>` is not implemented for `String`)
```

plus two warnings (`unused import: self` at `lib.rs:24`, `unused import: Manager` at `app_menus.rs:9`). Details inline.

## Why CI told you this was fine

Our `ci.yml` had **no cargo step at all** — the suite, the wrangler dry-run and the worker bundle are all JavaScript. So the Tauri crate could stop building and the checkmark stayed green. The only `cargo test` lived in `installer-release.yml`, which fires on an `installer-v*` tag, meaning this would have surfaced at release time.

That's fixed now (`894595b` on `v2.2`): CI has a `desktop` job running `cargo check`, and the `pull_request` branch filter is gone so every PR is checked regardless of base. Once you push a fix you should get a real signal instead of a misleading green.

## Smaller notes

- **No tests for ~1,175 new lines.** `i18n.rs` in particular is very testable — a locale-resolution test and a "every `Key` has a string in every locale" test would be cheap and would catch drift as locales are added.
- **Not your problem, just a heads-up:** this PR shifts line anchors that two open specs cite — #246 references `windows.rs:13` (now 15) and #247 references `commands.rs:275` (now 297) and `main.ts:459` (now 428). The code all still exists, it just moved. I'll update those issues.

## Base branch

I retargeted this from `main` to `v2.2` earlier, where the next release is assembling. The diff was unchanged by that and it still merges cleanly.

Genuinely sorry the CI gap sent you down this path — the fix is now in place, and the work itself looks good. Ping me when you've pushed and I'll re-review promptly.

**Comment by FaberVi** [comment]
Thanks for the careful review, @rahilp — and for fixing the CI gap so this would have been caught earlier.

Pushed `c73f18a` (plus merge of `v2.2`) addressing the three `cargo check` failures:

1. **`lib.rs`** — `use i18n::{AppLocale, Key}` (dropped `self`)
2. **`app_menus.rs`** — `install_tray` bound is now `Fn(&AppHandle, MenuEvent)` (owned event); removed unused `Manager`
3. **`commands.rs`** — `rebuild_tray_menu(...).map_err(|e| e.to_string())?`

Also added i18n tests for `resolve_initial_locale` and non-empty EN/IT strings for every `Key`.

Local `cargo` isn't on this machine's PATH, so relying on the new CI `desktop` job for the compile signal. Ready for re-review.

File: installer/src-tauri/src/lib.rs (hunk 1 above)
Comment (FaberVi) [reply installer/src-tauri/src/lib.rs]: Fixed — now `use i18n::{AppLocale, Key};` so `mod i18n` is the only binding.

File: installer/src-tauri/src/app_menus.rs (hunk 2 above)
Comment (FaberVi) [reply installer/src-tauri/src/app_menus.rs:263]: Fixed — bound is `Fn(&AppHandle, tauri::menu::MenuEvent)` (owned `MenuEvent`). Also dropped unused `Manager`.

File: installer/src-tauri/src/commands.rs (hunk 3 above)
Comment (FaberVi) [reply installer/src-tauri/src/commands.rs]: Fixed — `menus.rebuild_tray_menu(&app).map_err(|e| e.to_string())?;` Scanned nearby `?` sites; this was the only bare `tauri::Error` into `Result<_, String>`.

File: installer/src-tauri/src/app_menus.rs [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    tray_menu: &Menu<tauri::Wry>,
+    on_menu_event: F,
+) -> tauri::Result<()>
+where
+    F: Fn(&AppHandle, tauri::menu::MenuEvent) + Send + Sync + 'static,
+{
+    TrayIconBuilder::with_id("second-brain-tray")
+        .icon(
+            app.default_window_icon()
+                .expect("bundled icon")
```
Comment (rahilp) [inline installer/src-tauri/src/app_menus.rs:258]: Non-blocking question: this panics at startup if `default_window_icon()` is ever `None`.

In practice the icon is bundled through `tauri.conf.json`, so `None` means a broken build rather than a runtime condition — and failing loudly is arguably the right response to that. So I'm not asking you to change it.

The thing that gives me slight pause is that it's a panic in a shipped desktop app, on the launch path, where the user sees a crash rather than a message. If you'd rather degrade, `.ok_or(...)` and skipping the tray icon would leave the app usable.

Your call — happy either way, just flagging that the failure mode is "app won't start" rather than "tray has no icon".

File: installer/src-tauri/src/app_menus.rs [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+    pub tray_hub: MenuItem<tauri::Wry>,
+    pub tray_sync: MenuItem<tauri::Wry>,
+    pub tray_update: MenuItem<tauri::Wry>,
+    pub tray_logout: MenuItem<tauri::Wry>,
+    pub tray_quit: MenuItem<tauri::Wry>,
+}
+
+impl AppMenus {
+    pub fn apply_locale(&self, locale: Locale) {
+        let _ = self.menu_open.set_text(i18n::t(locale, Key::MenuOpenDashboard));
```
Comment (rahilp) [inline installer/src-tauri/src/app_menus.rs:29]: Minor: this block discards the result of every `set_text`, so a locale switch that partially fails leaves a half-translated menu with nothing logged.

Best-effort is a reasonable choice for UI updates and I wouldn't block on it — a failed `set_text` is rare. But with five or six in a row, a partial switch would be genuinely confusing to debug, since the user sees mixed languages and there's no signal anywhere.

A single `if let Err(e) = ... { log }` on the first failure, or collecting into a count, would be enough to make it diagnosable without changing the behaviour.

File: .gitignore [hunk 6]
```diff
@@ -6,6 +6,7 @@ coverage/
 .wrangler-dry-run/
 docs/
 .claude/
+.cursor/
```
Comment (rahilp) [inline .gitignore:9]: Trivial, and I nearly didn't mention it: `.cursor/` is one specific editor's directory, and it's unrelated to native menus and i18n.

The repo does already ignore `.DS_Store`, so personal artifacts aren't foreign here, and I'm not going to make an issue of one line. The usual argument for keeping editor dirs in a personal global gitignore (`~/.config/git/ignore`) is just that the shared file otherwise accumulates an entry per contributor per tool.

Entirely your preference — happy to merge as-is.

**Review by rahilp — APPROVED** [review]
Fixed, and nicely — approving. Thank you for turning this around so quickly, @FaberVi.

All three compile errors are corrected exactly at the root rather than papered over: the `self` dropped from the `i18n` import, the `&` removed from the `MenuEvent` bound, and `map_err(|e| e.to_string())` on `rebuild_tray_menu`. Minimal diffs, no collateral.

**Verification I ran against current `v2.2`** (which now carries the #245 config layer — your branch already has it merged in):

| Check | Result |
|---|---|
| Merge into `v2.2` | ✅ clean, no conflicts |
| `cargo check` | ✅ clean |
| `cargo test` | ✅ 53 passed, 0 failed |
| Worker suite | ✅ 826 tests, unaffected |
| `npm run typecheck` | ✅ 0 errors |
| installer `bundle-worker` + `vite build` | ✅ both green |
| CI (test + macOS + Windows) | ✅ all three |

**The i18n design is better than I gave it credit for in my last review.** I assumed `all_keys()` being a hand-maintained list meant a 67th key could slip through untranslated. It can't — `t()` is an exhaustive `match (Locale, Key)` with no catch-all, so adding a variant is a **compile error** naming both the `En` and `It` arms. I verified it by adding a key: `error[E0004]: non-exhaustive patterns: (Locale::En, Key::BrandNewUntranslatedKey) and (Locale::It, ...) not covered`. That's a stronger guarantee than any test.

The TypeScript side has the same property — `en.ts` and `it.ts` are both typed `Messages`, so adding a key to the type fails typecheck in *both* files until translated. I confirmed that too. Nice that the two layers are consistent.

**I mutation-tested the new i18n tests — 6/6 caught:** Italian string emptied, Italian silently replaced with the English text, `parse()` no longer recognising `it`, `parse()` losing its `to_lowercase`, `t_fmt` no longer substituting placeholders, and stored locale no longer preferred over system. They earn their place; `italian_menu_strings` asserting exact strings is what catches the "silently replaced with English" case, which is the realistic regression.

Three small notes inline, none blocking, all fine to ignore or handle later.

One coordination question that isn't about this code: #242 also adds a credits surface (`public/credits.js` plus CSS, on the dashboard). Different layer, no file conflict, and I've verified they don't collide — but are these meant to be the same feature in two places, or deliberately separate? Only asking so the copy doesn't drift apart later.

Once this lands it unblocks #246 and #247, both of which build directly on `windows.rs`, `commands.rs` and the i18n layer you've added here. Thanks again.

**Comment by rahilp** [comment]
Merged — thank you, @FaberVi.

Three surgical fixes, and the i18n design enforces itself: `t()` is an exhaustive match, so a missing translation is a compile error rather than a runtime surprise. Same guarantee on the TypeScript side via the shared `Messages` type. I mutation-tested the new tests — six for six caught, including an Italian string silently swapped for the English one.

Worth saying: our CI could not have caught those compile errors when you first pushed. There was no `cargo` step at all, so the crate could stop building while the checkmark stayed green. That gap was ours, and it is fixed now.

This unblocks #246 and #247. Thanks for the quick turnaround.
